static/app/utils/useApiRequests.tsx

```diff
@@ -305,69 +209,125 @@ function useApiRequests({
           isLoading,
           isReloading: prevState.isReloading && isLoading,
         };
-        markShouldMeasure({remainingRequests: newState.remainingRequests, error: true});
+        triggerMeasurement({finished: newState.remainingRequests === 0, error: true});
         return newState;
       });
 
       onRequestError(error, args);
     },
-    [markShouldMeasure, onRequestError]
+    [triggerMeasurement, onRequestError]
   );
 
-  async function fetchData(extraState: Partial<State> = {}) {
-    if (!endpoints.length) {
+  const fetchData = useCallback(
+    async (extraState: Partial<State> = {}) => {
+      // Nothing to fetch if enpoints are empty
+      if (!endpoints.length) {
+        setState(prevState => ({
+          ...prevState,
+          data: {},
+          isLoading: false,
+          hasError: false,
+        }));
+
+        return;
+      }
+
+      // Cancel any in flight requests
+      api.clear();
+
       setState(prevState => ({
         ...prevState,
-        isLoading: false,
+        isLoading: true,
         hasError: false,
+        remainingRequests: endpoints.length,
+        ...extraState,
       }));
-      return;
+
+      await Promise.all(
+        endpoints.map(async ([stateKey, endpoint, parameters, options]) => {
+          options = options ?? {};
+          // If you're using nested async components/views make sure to pass the
+          // props through so that the child component has access to props.location
+          const locationQuery = (location && location.query) || {};
+          let query = (parameters && parameters.query) || {};
+          // If paginate option then pass entire `query` object to API call
+          // It should only be expecting `query.cursor` for pagination
+          if ((options.paginate || locationQuery.cursor) && !options.disableEntireQuery) {
+            query = {...locationQuery, ...query};
+          }
+          try {
+            const results = await api.requestPromise(endpoint, {
+              method: 'GET',
+              ...parameters,
+              query,
+              includeAllArgs: true,
+            });
+            const [data, _, resp] = results;
+            handleRequestSuccess({stateKey, data, resp}, true);
+          } catch (error) {
+            handleError(error, [stateKey, endpoint, parameters, options]);
+          }
+        })
+      );
+    },
+    [api, endpoints, handleError, handleRequestSuccess, location]
+  );
+
+  const reloadData = useCallback(() => fetchData({isReloading: true}), [fetchData]);
+
+  const handleMount = useCallback(async () => {
+    try {
+      await fetchData();
+    } catch (error) {
+      setState(prevState => ({...prevState, hasError: true}));
+      throw error;
     }
+  }, [fetchData]);
 
-    // Cancel any in flight requests
-    api.clear();
+  // Trigger fetch on mount
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useEffect(() => void handleMount(), []);
```

Comment:
```suggestion
 useEffect(() => void handleMount(), [handleMount]);
```

Reply:
I actually think this may be a little unsafe. We really only want the requests to fire on mount, in the case where `fetchData` changes we would re trigger this, which may be undesirable